### PR TITLE
Allow logback nats appender transport

### PR DESCRIPTION
**Pull request by lucamilanesio**
Add an optional dependency to the forthcoming plugin that includes
    the LOGBack appender to push events to the NATS pub/sub broker.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -106,13 +107,36 @@
             <artifactId>jackson-databind</artifactId>
             <version>2.7.4</version>
         </dependency>
+        <dependency>
+            <groupId>ch.qos.logback</groupId>
+            <artifactId>logback-classic</artifactId>
+            <version>1.2.3</version>
```
Comment (oleg-nenashev) [inline pom.xml]: Do you want to have it as a mandatory dependency?
If it's just for logback util, there are ways to make it optional. But no strong opinion

Comment (lucamilanesio) [reply pom.xml]: True, I can define this as optional dependency and do a soft-fail.

Comment (lucamilanesio) [reply pom.xml]: Done, thanks @oleg-nenashev  for your feedback, good catch.

**Comment by maximecharron** [comment]
I think this PR needs to be rebased once the other PRs are merged. There seems to be more going on in this that what the titles of the PR implies.

**Comment by lucamilanesio** [comment]
Yes, let me rebase it.

**Comment by maximecharron** [comment]
I'd rebase after PR #9 is merged since you are upgrading to jenkins 2.0 in this PR.

**Comment by lucamilanesio** [comment]
Yep, releasing 1.1.3 with all the fixes that are compatible with Jenkins 1.x and Java 7 and then the first to go will be the Jenkins 2.0 + major version change to 2.0-SNAPSHOT

**Comment by lucamilanesio** [comment]
Abandoning this PR, going to squash them with the other related to NATS

**Pull request closed without merging**